src/transactionComputer.ts

```diff
@@ -57,6 +68,20 @@ export class TransactionComputer {
         return Buffer.from(hash, "hex");
     }
 
+    hasOptionsSetForGuardedTransaction(transaction: ITransaction): boolean {
+        return (transaction.options & TRANSACTION_OPTIONS_TX_GUARDED) == TRANSACTION_OPTIONS_TX_GUARDED;
+    }
+
+    hasOptionsSetForHashSigning(transaction: ITransaction): boolean {
+        return (transaction.options & TRANSACTION_OPTIONS_TX_HASH_SIGN) == TRANSACTION_OPTIONS_TX_HASH_SIGN;
+    }
+
+    applyGuardian(transaction: ITransaction, guardian: string) {
+        transaction.version = 2;
+        transaction.options = transaction.options | TRANSACTION_OPTIONS_TX_GUARDED;
+        transaction.guardian = guardian;
+    }
+
```

Comment:
Nice! We should also think of an `apply*` function for setting `TRANSACTION_OPTIONS_TX_HASH_SIGN`.